OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameMenuLogic.cs

```diff
@@ -100,50 +138,39 @@ public IngameMenuLogic(Widget widget, World world, Action onExit, WorldRenderer
 			};
 			surrenderButton.IsDisabled = () => world.LocalPlayer == null || world.LocalPlayer.WinState != WinState.Undefined;
 
-			var saveMapButton = menu.Get<ButtonWidget>("SAVE_MAP");
-			saveMapButton.IsVisible = () => world.Type == WorldType.Editor;
-			saveMapButton.OnClick = () =>
+			var restartButton = menu.Get<ButtonWidget>("RESTART");
+			restartButton.IsDisabled = () => !world.LobbyInfo.IsSinglePlayer;
+			Action restartMission = () =>
 			{
-				hideMenu = true;
-				var editorActorLayer = world.WorldActor.Trait<EditorActorLayer>();
-				Ui.OpenWindow("SAVE_MAP_PANEL", new WidgetArgs()
+				var difficulty = world.LobbyInfo.GlobalSettings.Difficulty;
+				OrderManager om = null;
+				Action lobbyReady = null;
+				lobbyReady = () =>
 				{
-					{ "onSave", (Action<string>)(_ => hideMenu = false) },
-					{ "onExit", () => hideMenu = false },
-					{ "map", world.Map },
-					{ "playerDefinitions", editorActorLayer.Players.ToMiniYaml() },
-					{ "actorDefinitions", editorActorLayer.Save() }
-				});
-			};
+					if (difficulty != null)
+						om.IssueOrder(Order.Command("difficulty {0}".F(difficulty)));
 
-			menu.Get<ButtonWidget>("MUSIC").OnClick = () =>
-			{
-				hideMenu = true;
-				Ui.OpenWindow("MUSIC_PANEL", new WidgetArgs()
-				{
-					{ "onExit", () => hideMenu = false },
-					{ "world", world }
-				});
+					Game.LobbyInfoChanged -= lobbyReady;
+					om.IssueOrder(Order.Command("state {0}".F(Session.ClientState.Ready)));
+				};
+
+				Game.Disconnect();
+				//Ui.ResetAll();
+
+				Game.LobbyInfoChanged += lobbyReady;
+				om = Game.JoinServer(IPAddress.Loopback.ToString(), Game.CreateLocalServer(world.Map.Uid), "", false);
```

Comment:
What should be changed here so that it writes to a new replay file for skirmish games?